src/misc/support.cpp

```diff
@@ -124,6 +124,36 @@ void trim(std::string &str)
 	str.erase(0, empty_pfx);
 }
 
+std::vector<std::string> split(const std::string &seq, const char delim)
```

Comment:
Both versions of this function behave better now - thanks :)

Only now I recognized a problem with the version that's not using delim char - we cannot easily determine vector size to reserve before filling it. The ways to *not* solve this:

1. Do NOT use std::list

Three possible ways to address this:

1. We could use std::forward_list - not the perfect solution, but OK, and lower overhead than std::list
2. We could use std::deque - faster than std::list, but with high overhead for cases with small number of elements
3. Calculate the number of elements to reserve for std::vector. For function using delimiter it's simple - just count the number of delimiters in input string (single pass through the string is less work than several re-initializations of std::vector's backing buffer). For function splitting by space it's more complicated - we could calculate number of adjacent space/non-space characters… but maybe it's an overengineering at this point of implementation - it will be much easier to initially assume that the majority of `split` calls will generate small outputs and reserve something reasonable - e.g. start with reserving 5 elements.

Reply:
Thanks for this feedback @dreamer. I also like the count and reserve approach, given these functions might wind up being used at runtime too where performance becomes more important. Will take a crack at the run-length counts on the white space splitter; it shouldn't be too onerous.

Reply:
Without actually running through the compiler - adjacent chars counter could look something like this:
```C++
size_t n = 1;
while (std::adjacent_find(begin(str), end(str), [](char a, char b) { return !isspace(a) && isspace(b); } != end(str))
        n += 1;
```

Reply:
That one repeats the same search (so spins).  

This handles all the corner cases, like `" a b"`, `"a b "`, and `" a b "`.  The assignment of `head = ` and comparison to `seq.end()` can all be done inside the `while`, but it's borderline obfuscated at that point.

``` c++
    auto head = seq.begin();
    size_t n = !isspace(seq.back());
    while (head != seq.end()) {
        head = std::adjacent_find(head,  seq.end(), [](char a, char b) { return !isspace(a) && isspace(b); });
        if (head != seq.end()) {
            ++n;
            ++head;
        }
    }
```

Reply:
I just went with the original pass through, but turned it into a count:
``` c++
	size_t n = 0;
	auto head = seq.find_first_not_of(whitespace, 0);
	while (head != std::string::npos) {
		const auto tail = seq.find_first_of(whitespace, head);
		head = seq.find_first_not_of(whitespace, tail);
		++n;
	}
```

Reply:
Latest update asserts that the calculated number of reserved elements matches the actual elements populating the vector.